homeassistant/components/plugwise/climate.py

```diff
@@ -0,0 +1,250 @@
+        attributes["previous_schema"] = self._previous_schema
+        return attributes
+
+    def update(self):
+        """Update the data from the thermostat."""
+        _LOGGER.debug("Update called")
+        self._domain_objects = self._api.get_domain_objects()
+        self._outdoor_temperature = self._api.get_outdoor_temperature(
+            self._domain_objects
+        )
+        api_active_schema = self._api.get_active_schema_name(self._domain_objects)
+        if self._active_schema != api_active_schema:
+            self._previous_schema = self._active_schema
+            self._active_schema = api_active_schema
+
+    @property
+    def hvac_mode(self):
+        """Return current active hvac state."""
+        if self._api.get_schema_state(self._domain_objects) is True:
+            return HVAC_MODE_AUTO
+        return HVAC_MODE_OFF
+
+    @property
+    def preset_mode(self):
+        """Return the active preset mode."""
+        return self._api.get_current_preset(self._domain_objects)
+
+    @property
+    def preset_modes(self):
+        """Return the available preset modes list without values."""
+        presets = list(self._api.get_presets(self._domain_objects).keys())
+        return presets
+
+    @property
+    def hvac_modes(self):
+        """Return the available hvac  modes list."""
+        return self._hvac_modes
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._api.get_temperature(self._domain_objects)
+
+    @property
+    def min_temp(self):
+        """Return the minimal temperature possible to set."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature possible to set."""
+        return self._max_temp
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._api.get_target_temperature(self._domain_objects)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measured temperature."""
+        return TEMP_CELSIUS
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        _LOGGER.debug("Adjusting temperature")
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is not None and self._min_temp < temperature < self._max_temp:
+            self._temperature = temperature
+            self._api.set_temperature(self._domain_objects, temperature)
+            _LOGGER.debug("Changing temporary temperature")
+        else:
+            _LOGGER.error("Invalid temperature requested")
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set the hvac mode."""
+        _LOGGER.debug("Adjusting hvac_mode (i.e. schedule/schema)")
+        if self._previous_schema is None and self._active_schema is None:
+            _LOGGER.error("previous_schema not known, unable to apply")
+        else:
+            if self._previous_schema is None:
```

Comment:
Is it correct that we should use the previous schema and fall back to the active schema if previous schema is missing? I would have assumed it should be the opposite. Of course I don't know this device.

Reply:
This was trying to solve a 'catch 22' scenario but due to the other improvements through your review critics we can clean this up completely